### DISCUSSION TITLE
Binding issue about NonReloadingTabControl.cs

### DISCUSSION THREAD

**Discussion by NeverMorewd** [cefsharp/CefSharp#4583] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

108.4.13

### Operating System

Windows 11

### Architecture

x64

### .Net Version

.Net 6

### Implementation

WPF

### Reproduction Steps

CefSharp.Wpf.Example is a good sample for learning how to use cefsharp in wpf.I noticed that NonReloadingTabControl was considered. There is no problem in CefSharp.Wpf.Example.However when I try to access NonReloadingTabControl  with **ReactiveUI-Mvvm**,I encountered a issue.

The following codes for reference.

#### viewmode.cs:
```
 private readonly ReadOnlyObservableCollection<MockModel> _mocks;

 public MainWindowViewModel()
 {
      var mockLoadCacheCleanup = mockService.All.Connect()
                  .Transform(t => t)
                  .ObserveOn(schedulerProvider.MainThread)
                  .Bind(out _mocks)
                  .DisposeMany()
                  .Subscribe();
 }

 public ReadOnlyObservableCollection<MockModel> Mocks => _mocks;
```

#### view.xaml:

 ``` 
<theme:NonReloadingTabControl
                            HorizontalContentAlignment="Left"
                            BorderBrush="#333333"
                            BorderThickness="0"
                            ItemsSource="{Binding Mocks}"/>
```

**When I add the first MockModel to _mocks,The NonReloadingTabControl just show the TabItem-Header,the TabItem-Content was collapsed.**

I looked into the code in NonReloadingTabControl.cs
The first adding to _mocks triggered OnItemsChanged of NonReloadingTabControl,And the e.Action is Reset.So the itemsHolderPanel.Children.Clear(); was called.

 ``` 
protected override void OnItemsChanged(NotifyCollectionChangedEventArgs e)
{
            base.OnItemsChanged(e);

            if (itemsHolderPanel == null)
            {
                return;
            }

            switch (e.Action)
            {
                case NotifyCollectionChangedAction.Reset:
                    **itemsHolderPanel.Children.Clear();**
                    break;
 ``` 

Actually, this is not the issue of CefSharp.But if it can be fixed, it will give developers better guidance.

### Expected behavior

When the Firstly adding to ReadOnlyObservableCollection,The TabControl can show header and content normally.

### Actual behavior

When the Firstly adding to ReadOnlyObservableCollection,The TabControl  show header only,content is missed.

### Known Workarounds

Currently,I go with a workaround that delete switch (e.Action) code block.Reserve base.OnItemsChanged(e); only.
Hostly, removing the overriding to OnItemsChanged will work as well.

### Does this problem also occur in the CEF Sample Application

Yes using WPF/OffScreen command line args

**Comment by amaitland** [comment]
Converted to discussion as this isn't a bug in `CefSharp`.

> Actually, this is not the issue of CefSharp.But if it can be fixed, it will give developers better guidance.

As per the xml doc `NonReloadingTabControl` was adapted from http://stackoverflow.com/a/9802346

It's not something that was created as part of this project. I'm not sure what's required to support other frameworks like `ReactiveUI-Mvvm`. Trying to provide generic support for a control that is provided purely as part of an example isn't really practical. There's no way to test or debug other frameworks.

I haven't searched for other alternative solutions in a long time, there are likely others. For reference see https://github.com/cefsharp/CefSharp/discussions/3862#discussioncomment-1548284

If someone wants to create (or knows of) an open source project that provides a non virtualising tab control then I'm happy to switch the example to using that.
